storage/options.go

```diff
@@ -138,29 +162,63 @@ func NewOptions() Options {
 		errThresholdForLoad:            defaultErrorThresholdForLoad,
 		repairEnabled:                  defaultRepairEnabled,
 		repairOpts:                     repair.NewOptions(),
-		fileOpOpts:                     NewFileOpOptions(),
 		bootstrapProcess:               defaultBootstrapProcess,
 		persistManager:                 fs.NewPersistManager(fs.NewOptions()),
+		tickInterval:                   defaultTickInterval,
 		maxFlushRetries:                defaultMaxFlushRetries,
-		contextPool:                    context.NewPool(nil, nil),
-		seriesPool:                     series.NewDatabaseSeriesPool(series.NewOptions(), nil),
+		poolOpts:                       poolOpts,
+		contextPool:                    context.NewPool(poolOpts, poolOpts),
+		seriesOpts:                     seriesOpts,
+		seriesPool:                     series.NewDatabaseSeriesPool(seriesOpts, poolOpts),
 		bytesPool:                      bytesPool,
-		encoderPool:                    encoding.NewEncoderPool(nil),
-		segmentReaderPool:              xio.NewSegmentReaderPool(nil),
-		readerIteratorPool:             encoding.NewReaderIteratorPool(nil),
-		multiReaderIteratorPool:        encoding.NewMultiReaderIteratorPool(nil),
-		identifierPool:                 ts.NewIdentifierPool(bytesPool, nil),
-		fetchBlockMetadataResultsPool:  block.NewFetchBlockMetadataResultsPool(nil, 0),
-		fetchBlocksMetadataResultsPool: block.NewFetchBlocksMetadataResultsPool(nil, 0),
+		encoderPool:                    encoding.NewEncoderPool(poolOpts),
+		segmentReaderPool:              xio.NewSegmentReaderPool(poolOpts),
+		readerIteratorPool:             encoding.NewReaderIteratorPool(poolOpts),
+		multiReaderIteratorPool:        encoding.NewMultiReaderIteratorPool(poolOpts),
+		identifierPool:                 ts.NewIdentifierPool(bytesPool, poolOpts),
+		fetchBlockMetadataResultsPool:  block.NewFetchBlockMetadataResultsPool(poolOpts, 0),
+		fetchBlocksMetadataResultsPool: block.NewFetchBlocksMetadataResultsPool(poolOpts, 0),
 	}
 	return o.SetEncodingM3TSZPooled()
 }
 
+func (o *options) Validate() error {
+	if o.tickInterval <= 0 {
+		return errTickIntervalNegative
```

Comment:
I think we should actually be checking for a positive number, we need tick > 0 or else we're fast loop ticking.